### PR TITLE
File upload status

### LINKED ISSUE #121: Indicate upload status for file upload
Currently there is no indicator of  how much has been uploaded or ETA of the file upload.

### PR DESCRIPTION
**Pull request by romancow**
This keeps the upload dialog open during the entire upload and shows a linear progress indicator underneath each file being uploaded.

I also moved the "Append Arttifact" dialog out from `App.svelte` into its own component. The App component is getting pretty large and unwieldy, and this would add more.

Screen Shot 2023-01-26 at 4 44 24 PM

@brollb @yogeshVU 

resolves #121

### CODE REVIEW

File: src/routers/Search/dashboard/src/components/AppendArtifactDialog.svelte [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+    {#if !uploading}
+      <Dropzone on:drop={onAppendFileDrop} multiple={true}>
+        <p>Select dataset to upload.</p>
+      </Dropzone>
+    {:else}
+      <Dropzone disabled multiple={true}>
+        <p>Select dataset to upload.</p>
+      </Dropzone>
+    {/if}
```
Comment (romancow) [inline src/routers/Search/dashboard/src/components/AppendArtifactDialog.svelte:185]: fyi, this `if`/`then` for `Dropzone` was a workaround for a fact that the component's `disabled` property doesn't appear to be reactive. It probably would be preferable to patch the package to make it reactive, but didn't want to mess with that at this point.

**Comment by yogeshVU** [comment]
Tried running it locally, the dialog box does not support uploading multiple files when uploading. It however, uploaded a single file successfully.

**Comment by yogeshVU** [comment]
lgtm, we can create a separate issue for selecting multiple files for different select file dialog boxes

**Review by yogeshVU — APPROVED** [review]
lgtm

**Review by brollb — APPROVED** [review]

**Comment by brollb** [comment]
Looks good. Nice work, @romancow!

**Pull request merged**